src/formatDate.js

```diff
@@ -50,7 +50,47 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const arrDate = date.split(fromFormat[3]);
+
+  const outputDate = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const inputPart = toFormat[i];
+    let year;
+
+    const formatIndex = fromFormat.indexOf(inputPart);
+    const part = arrDate[formatIndex];
+
+    if (part === undefined) {
+      switch (inputPart) {
+        case 'YY':
+          year = arrDate[fromFormat.indexOf('YYYY')].slice(-2);
+
+          outputDate.push(year);
+          break;
+
+        case 'YYYY':
+          year = arrDate[fromFormat.indexOf('YY')];
+
+          // Перевірка, чи 'YY' < 30 та 'YY' = 00
+          if (year < 30 || year === '00') {
+            year = '20' + year;
+          } else {
+            year = '19' + year;
+          }
```

Comment:
let's move getting year logic to a separate helper in order to keep our cod clean and readable